src/hooks/context/useAccountsContextSetUp.ts

```diff
@@ -54,11 +56,15 @@ const useAccountsContextSetUp = () => {
   useEffect(() => {
     if (keyringPairsReady && keyringPairs.length) {
       dispatchAccount(AccountActionCreators.setAccounts(keyringPairs));
-      // This initial load might be temporary or it might change based on what it was disscussed in issue #224
       updateSenderAccountsInformation(dispatchAccount);
     }
   }, [keyringPairsReady, keyringPairs, dispatchAccount, updateSenderAccountsInformation]);
 
+  useEffect(() => {
+    const id = setInterval(() => updateSenderAccountsInformation(dispatchAccount), informationUpdateFrequency);
+    return () => clearInterval(id);
```

Comment:
This is not the best way to implement intervals cause it doesn't take into account delay caused by the function itself or handles errors. With low frequencies (or larger number of clients) that's basically a recipe to kill (ddos) your own server.

Imagine this:
1. frequency is low (say 3 seconds)
2. Response time is high (say 5 seconds)

This code will keep spamming node even if the node is not able to complete previous queries. With frequency higher than response rate it will just start eat up memory and eventually crash.

A better way to implement such refreshing is roughly like this:
```js
function updateMe(numberOfFailures) {
  updateSomethingAsync()
     .then(() => setTimeout(() => updateMe(0), DEFAULT_DELAY))
     .catch(e => {
       // since we failed, we are going to do exponential back-off to not overload the server.
       const delay = Math.min(MAX_DELAY, DEFAULT_DELAY * Math.pow(2, numberOfFailures));
       setTimeout(() => updateMe(numberOfFailures + 1), delay);
    });
}
```

Reply:
i don't have anything to say about this.

Reply:
I have - you need to make sure that you will clear the `setTimeout` during unmounting of the hook. 
Else you can have red screens and errors

Reply:
Yes, my submitted work handles the the `clearInterval` in the useEffect cleanup.

Reply:
@tomusdrw if we implement the block number count + 30 seconds (whatever it comes first and after an update process, counters + timer gets reseted ), do we need to handle the catch error scenario? I understand your point and it is pretty valid, but i feel that is too much for what we are doing here, since if we need low frequency update we would just leave the balances update using subscriptions as it used to be before. The idea of this change is to update balances with  a larger amount of time ( around 4 blocks or 30 seconds ).

Btw ... do you want to establish these two numbers for now? 4 blocks  and 30 seconds?

Reply:
> Yes, my submitted work handles the the `clearInterval` in the useEffect cleanup.

Im not talking about the current code, but the code that Tomasz proposes.